connectors/sources/aws.py

```diff
@@ -3,116 +3,223 @@
 # or more contributor license agreements. Licensed under the Elastic License 2.0;
 # you may not use this file except in compliance with the Elastic License 2.0.
 #
-import os
-from hashlib import md5
 import asyncio
-from functools import partial
 import logging
+import os
+from functools import partial
+from hashlib import md5
 
 import aioboto3
+from aiobotocore.response import AioReadTimeoutError
+from aiohttp.client_exceptions import ServerTimeoutError
+from aiobotocore.config import AioConfig
 from aiobotocore.utils import logger as aws_logger
-
-from connectors.source import BaseDataSource
+from botocore.exceptions import ClientError
 from connectors.logger import logger, set_extra_logger
-
+from connectors.source import BaseDataSource
 
 SUPPORTED_CONTENT_TYPE = [
     "text/plain",
 ]
 SUPPORTED_FILETYPE = [".py", ".rst", ".rb", ".sh", ".md", ".txt"]
 ONE_MEGA = 1048576
+ENDPOINT_URL = os.getenv(
+    key="ENDPOINT_URL", default=None
+)  # For end to end tests set this environment variable with target host.
 
 
 class S3DataSource(BaseDataSource):
     """Amazon S3"""
 
     def __init__(self, connector):
+        """Setup connection to the Amazon S3.
+
+        Args:
+            connector (BYOConnector): Object of the BYOConnector class.
+        """
         super().__init__(connector)
         self.session = aioboto3.Session()
         self.loop = asyncio.get_event_loop()
         set_extra_logger(aws_logger, log_level=logging.INFO, prefix="S3")
```

Comment:
we might want to set the log level to DEBUG here

Reply:
@tarekziade We checked setting the log_level as DEBUG here, but in that case suppose we have a bucket in region "us-east-1" and aws configuration default region is "ap-south-1", so we are getting a debug log for all the object exist in that bucket which says:
S3 request was previously redirected, not redirecting.

> [S3][10:26:51][DEBUG] S3 client configured for region ap-south-1 but the bucket bucket-50k is in region us-east-1; Please configure the proper region to avoid multiple unnecessary redirects and signing attempts.

Do you think it's okay to print so many logs for DEGUB?

Reply:
@tarekziade I set the log_level as DEBUG here, but in that case suppose we have a bucket in region "us-east-1" and aws configuration default region is "ap-south-1", so we are getting a debug log for all the object exist in that bucket which says:
S3 request was previously redirected, not redirecting.

> [S3][10:26:51][DEBUG] S3 client configured for region ap-south-1 but the bucket bucket-50k is in region us-east-1; Please configure the proper region to avoid multiple unnecessary redirects and signing attempts.

Do you think it's okay to print so many logs for DEGUB?